hydromt/io.py

```diff
@@ -226,6 +228,81 @@ def open_mfraster(
     return ds
 
 
+def open_mfcsv(
+    fns: Dict[Union[str, int], Union[str, Path]],
+    concat_dim: str,
+    driver_kwargs: Optional[Dict[str, Any]] = None,
```

Comment:
You can also initialise with an empty dict rather than None

Reply:
it is considered really bad practices to initialise default arguments with things like lists or dictionaries. (e.g. https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments) because it has really counterintuitive results. So unless there is a good reason for, I think it's better to keep the default as None.